src/app/FAKE/CommandlineParams.fs

```diff
@@ -4,18 +4,35 @@ open Fake
 
 let printAllParams() = printfn "FAKE.exe [buildScript] [Target] Variable1=Value1 Variable2=Value2 ... "
 
+(* 
+    This is a set of flags that exist in code lower in the target processing that MUST be normalized to a certain form. This is necessary because
+    If any old styole variables are present in the FAKE invocation, Argu parsing will fail and we will not have parsed out those commands correctly
+    from the overall command line.
+
+    You can typically find usages of 'hasBuildParam' in this codebase to find places where these values are required.
+*)
+let specialFlags = 
+    [
+        "-st", "single-target"
+        "--single-target", "single-target" 
+        "-pd", "details"
+        "--print-details", "details"
+    ] |> Map.ofList
+
 let parseArgs cmdArgs = 
-    let splitter = [| '=' |]
-    let split (arg:string) = 
-      let pos = arg.IndexOfAny splitter 
-      [| arg.Substring(0, pos); arg.Substring(pos + 1, arg.Length - pos - 1) |]
+    let (|KeyValue|Flag|TargetName|) ((i,arg) : int * string) =
+        if i = 0 then TargetName arg
+        else
+            match arg.IndexOf '=' with
+            | -1 -> Flag arg
+            | i -> KeyValue (arg.Substring(0, i), arg.Substring(i + 1, arg.Length - i - 1))
+
     cmdArgs
     |> Seq.skip 1
-    |> Seq.mapi (fun (i : int) (arg : string) -> 
-           if arg.Contains "=" then 
-               let s = split arg
-               if s.[0] = "logfile" then addXmlListener s.[1]
-               s.[0], s.[1]
-           else if i = 0 then "target", arg
-           else arg, "true")
+    |> Seq.mapi (fun i a -> match (i, a) with 
+                            | TargetName t -> "target", t
+                            | Flag f when Map.containsKey f specialFlags -> Map.find f specialFlags, "true"
```

Comment:
This is the only new case, really, where we expand out new-style flags that had special handling in the code (See Program.fs where we yield ('single-target', true) for the new-style args as an example).